event-handlers/src/test/java/no/sikt/nva/nvi/events/evaluator/EvaluateNviCandidateHandlerTest.java

```diff
@@ -592,11 +519,119 @@ void shouldHandleJournalWithMultipleTypes() throws IOException {
+                                                        URI publicationBucketUri) {
+        var verifiedCreators = List.of(new NviCreator(HARDCODED_CREATOR_ID, List.of(CRISTIN_NVI_ORG_SUB_UNIT_ID)));
+        return NviCandidate.builder()
+                   .withPublicationId(HARDCODED_PUBLICATION_ID)
+                   .withPublicationBucketUri(publicationBucketUri)
+                   .withDate(HARDCODED_PUBLICATION_DATE)
+                   .withInstanceType(instanceType)
+                   .withChannelType(channelType.getValue())
+                   .withLevel(level)
+                   .withPublicationChannelId(HARDCODED_PUBLICATION_CHANNEL_ID)
+                   .withIsInternationalCollaboration(false)
+                   .withCollaborationFactor(ONE.setScale(1, ROUNDING_MODE))
+                   .withCreatorShareCount(countCreatorShares(verifiedCreators))
+                   .withBasePoints(basePoints)
+                   .withNviCreators(verifiedCreators)
+                   .withInstitutionPoints(institutionPoints.entrySet().stream()
+                                              .map(entry -> new InstitutionPoints(entry.getKey(), entry.getValue(),
+                                                                                  List.of(
+                                                                                      new CreatorAffiliationPoints(
+                                                                                          HARDCODED_CREATOR_ID,
+                                                                                          CRISTIN_NVI_ORG_SUB_UNIT_ID,
+                                                                                          entry.getValue()))))
+                                              .toList())
+                   .withTotalPoints(totalPoints)
+                   .build();
+    }
+
+    private static int countCreatorShares(List<NviCreator> nviCreators) {
+        return (int) nviCreators.stream()
+                         .mapToLong(creator -> creator.nviAffiliations().size())
+                         .sum();
+    }
+
+    private URI setupPublicationWithInvalidYear(String year) throws IOException {
+        var path = "evaluator/candidate_publicationDate_replace_year.json";
+        var content = IoUtils.stringFromResources(Path.of(path)).replace("__REPLACE_YEAR__", year);
+        return s3Driver.insertFile(UnixPath.of(path), content);
+    }
+
+    private void persistPeriod(int publishingYear) {
+        periodRepository.save(DbNviPeriod.builder()
+                                  .publishingYear(String.valueOf(publishingYear))
+                                  .startDate(LocalDateTime.of(publishingYear, 4, 1, 0, 0, 0).toInstant(ZoneOffset.UTC))
+                                  .reportingDate(
+                                      LocalDateTime.of(publishingYear + 1, 3, 1, 0, 0, 0).toInstant(ZoneOffset.UTC))
```

Comment:
Is there some significance of these arcane integers? Is it something like "then plus some time"?

Reply:
Refactored